### PR TITLE
add generator arch and other realted modules

### PR DESCRIPTION
**Pull request by ShakoHo**
@MikeLien r?

### CODE REVIEW

File: lib/common/imageUtil.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+            int((ref_end_point + input_cover_sec) * input_fps)]
+    logger.info("Image Comparison search range [%s] when image list len is [%s]: " % (str(search_range), str(input_image_list_len)))
+    return search_range
+
+
+def parallel_compare_image(input_sample_data, input_image_data, input_settings, result_list):
+    # init value
+    sample_dct = None
+    image_fn_list = copy.deepcopy(input_image_data.keys())
+    image_fn_list.sort()
```
Comment (MikeLien) [inline lib/common/imageUtil.py]: sort file name with number need to consider string type of number
or the will be [img_1.bmp, img_1x.bmp, img_1xxx.bmp, img_2.bmp, img_2x.bmp, img_2xxx.bmp, etc.]
`from commonUtil import CommonUtil`
`image_fn_list.sort()` -> `image_fn_list.sort(key=CommonUtil.natural_keys)`

File: lib/common/imageUtil.py [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+    """
+    generate crop data for crop images functions
+    @param input_target_list: target obj list, for example {'fp': 'xxxx, 'write_to_file': True}
+    @param crop_taget_list: crop target area list, for example {'viewport': {'x': 2, 'y', ..}}
+    @return: integrate crop data, for example: {'fp_list': [ {'input_fp':xxx, 'output_fp':xxx}], 'crop_area': {'x':2, 'y':3, ..}}
+    """
+    crop_data_dict = {}
+    for input_target in input_target_list:
+        if 'write_to_file' in input_target:
+            if input_target['write_to_file'] is True:
```
Comment (ShakoHo) [inline lib/common/imageUtil.py]: can't combine these two lines into one line due to attribute could missing in input_target

File: lib/helper/generatorHelper.py [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+logger = get_logger(__name__)
+RET_SUCCESSS = 0
+
+DEFAULT_FPS_VALIDATOR_NAME = 'FPSValidator'
+DEFAULT_FILEEXIST_VALIDATOR_NAME = 'FileExistValidator'
+DEFAULT_DCTRUNTIME_GENERATOR_NAME = 'DctRunTimeGenerator'
+DEFAULT_FFMPEG_CONVERTER_NAME = 'FfmpegConverter'
+DEFAULT_CV2_CONVERTER_NAME = 'Cv2Converter'
+DEFAULT_SAMPLE_CONVERTER_NAME = 'SampleConverter'
+DEFAULT_IMG_CLEAVER_NAME = 'ImgCleaver'
```
Comment (MikeLien) [inline lib/helper/generatorHelper.py]: this is an unnecessary variable

**Comment by MikeLien** [comment]
lgtm r+

**Pull request merged**